MNE/deep/deep.h

```diff
@@ -236,12 +246,17 @@ class DEEPSHARED_EXPORT Deep
     */
     bool trainMinibatch(const Eigen::MatrixXf& input, const Eigen::MatrixXf& targets, double& loss, double& error, const CNTK::DeviceDescriptor& device = CNTK::DeviceDescriptor::DefaultDevice());
 
+    void cancelTraining();
+
     //=========================================================================================================
     /**
     * Print the model structure
     */
     void print();
 
+signals:
```

Comment:
And what also should be considered to have multiple public private sections, which are grouping certain functionalities together - at the end of each such group there should reside the respective signals